### ISSUE TITLE
Issue with new naming spec and config table editor

### ISSUE THREAD

**Issue by LVerhulst4321** [LVerhulst4321/PlanZ#39]
I have discovered that there is a bit of an issue between our new naming conventions for tables and with the config table editor. The config table editor has specific code regarding tables that have references to other tables (like RoomHasSet). Our new naming conventions doesn't follow the usual Zambia naming conventions and so the config table editor doesn't work for any new tables with references. As best as I can tell.

I would appreciate it if someone else can check this out and verify what I am thinking is the issue.

I've made a new branch available called: adjust-public-grid-report
The public grid report is something that I created years ago for Capricon and then shoehorned in for Zambia. It is not pretty, but is functional. I'm working (slowly) on making it better.
Anyway, I had added a field in the Rooms table for divisionid, but I have decided to do something different. I've made a table called room_report_group to identify a "report group" and then a table called room_report_group_has_room to link a set of rooms to a report group.
After updating the config table editor code, I find that I get an error when trying to edit the room_report_group_has_room table and I believe it's due to the naming convention differences.

**Comment by lostcarpark** [comment]
I've done some work on the configuration editor. I'm happy to have a look at it later this evening.

**Comment by lostcarpark** [comment]
The configuration editor uses the foreign key to look up refrenced tables and populate a drop down.
The problem was that the configuration editor was assuming that there would be a field called `something_id` and a matching field called `something_name`. It was using a string replacement to change `id` to `name` on the primary key field.
The new naming standard calls for the primary key field to be called `id`.
I don't like the idea of requiring the name field to be called `name` (which would work, but seems fragile).
Instead, I've added a function to the file called `lookupNameColumn`. This will find the first column ending in `name`. This should work for existing tables with the name column matching the key, but going forward will only require there be a column that ends with `name` on the table.
Currently, it won't fail cleanly if there is no `name` column. It may also give unexpected results if there are multiple columns that end with `name`.
I've created a merge request into the adjust-public-grid-report branch.
